src/daft-plan/src/builder.rs

```diff
@@ -175,30 +189,69 @@ impl LogicalPlanBuilder {
         Ok(logical_plan.into())
     }
 
-    pub fn project(
+    pub fn select(&self, to_select: Vec<ExprRef>) -> DaftResult<Self> {
+        err_if_agg!("project", &to_select);
+
+        let logical_plan: LogicalPlan =
+            logical_ops::Project::try_new(self.plan.clone(), to_select, Default::default())?.into();
+        Ok(logical_plan.into())
+    }
+
+    pub fn with_columns(
         &self,
-        projection: Vec<ExprRef>,
+        columns: Vec<ExprRef>,
         resource_request: ResourceRequest,
     ) -> DaftResult<Self> {
-        for expr in &projection {
-            if check_for_agg(expr) {
-                return Err(DaftError::ValueError(format!(
-                    "Aggregation expressions are not currently supported in projection: {expr}\nIf you would like to have this feature, please see https://github.com/Eventual-Inc/Daft/issues/1979#issue-2170913383"
-                )));
-            }
-        }
+        err_if_agg!("with_column", &columns);
+
+        let new_col_names = columns
+            .iter()
+            .map(|e| e.name())
+            .collect::<DaftResult<HashSet<&str>>>()?;
+
+        let mut exprs = self
+            .schema()
+            .fields
+            .iter()
+            .filter_map(|(name, _)| {
+                if new_col_names.contains(name.as_str()) {
+                    None
+                } else {
+                    Some(col(name.clone()))
+                }
+            })
+            .collect::<Vec<_>>();
+
+        exprs.extend(columns);
```

Comment:
Nit: we could just chain `exprs`: `exprs...filter_map(...).chain(columns.into_iter()).collect()`

Reply:
We decided that we could not do this because the borrow of `columns` is not released before it is being consumed in the chain, so we would need to do an extra clone